eidas-saml-parser/src/test/java/uk/gov/ida/notification/eidassaml/EidasSamlResourceTest.java

```diff
@@ -80,6 +83,70 @@ public void shouldReturnRequestIdAndIssuer() throws Exception {
+        final AuthnRequest eidasAuthnRequest = createEidasAuthnRequestBuilder().build();
+        NameIDPolicy nameIDPolicy = SamlBuilder.build(NameIDPolicy.DEFAULT_ELEMENT_NAME);
+        nameIDPolicy.setFormat(NameID.PERSISTENT);
+        eidasAuthnRequest.setNameIDPolicy(nameIDPolicy);
+
+        final CountryMetadataResponse countryMetadataResponse = createCountryMetadataResponse(eidasAuthnRequest, new AssertionConsumerService(UriBuilder.fromPath(TEST_CONNECTOR_DESTINATION).build(), 0, true));
+
+        when(mockMetatronProxy.getCountryMetadata(eidasAuthnRequest.getIssuer().getValue())).thenReturn(countryMetadataResponse);
+
+        final EidasSamlParserResponse response = postEidasAuthnRequest(eidasAuthnRequest);
+
+        assertThat(response.isTransientPid()).isEqualTo(false);
+    }
+
+    @Test
+    public void shouldReturnFalseWhenNameIdIsUnspecified() throws Exception {
+        final AuthnRequest eidasAuthnRequest = createEidasAuthnRequestBuilder().build();
+        NameIDPolicy nameIDPolicy = SamlBuilder.build(NameIDPolicy.DEFAULT_ELEMENT_NAME);
+        nameIDPolicy.setFormat(NameID.UNSPECIFIED);
+        eidasAuthnRequest.setNameIDPolicy(nameIDPolicy);
+
+        final CountryMetadataResponse countryMetadataResponse = createCountryMetadataResponse(eidasAuthnRequest, new AssertionConsumerService(UriBuilder.fromPath(TEST_CONNECTOR_DESTINATION).build(), 0, true));
+
+        when(mockMetatronProxy.getCountryMetadata(eidasAuthnRequest.getIssuer().getValue())).thenReturn(countryMetadataResponse);
+
+        final EidasSamlParserResponse response = postEidasAuthnRequest(eidasAuthnRequest);
+
+        assertThat(response.isTransientPid()).isEqualTo(false);
+    }
+
+    @Test
+    public void shouldReturnTrueWhenNameIdIsTransient() throws Exception {
+        final AuthnRequest eidasAuthnRequest = createEidasAuthnRequestBuilder().build();
+        NameIDPolicy nameIDPolicy = SamlBuilder.build(NameIDPolicy.DEFAULT_ELEMENT_NAME);
+        nameIDPolicy.setFormat(NameID.TRANSIENT);
+        eidasAuthnRequest.setNameIDPolicy(nameIDPolicy);
+
+        final CountryMetadataResponse countryMetadataResponse = createCountryMetadataResponse(eidasAuthnRequest, new AssertionConsumerService(UriBuilder.fromPath(TEST_CONNECTOR_DESTINATION).build(), 0, true));
+
+        when(mockMetatronProxy.getCountryMetadata(eidasAuthnRequest.getIssuer().getValue())).thenReturn(countryMetadataResponse);
+
+        final EidasSamlParserResponse response = postEidasAuthnRequest(eidasAuthnRequest);
+
+        assertThat(response.isTransientPid()).isEqualTo(true);
+    }
+
+    @Test
+    public void shouldReturnFalseWhenNameIdIsNull() throws Exception {
+        final AuthnRequest eidasAuthnRequest = createEidasAuthnRequestBuilder().build();
+        NameIDPolicy nameIDPolicy = SamlBuilder.build(NameIDPolicy.DEFAULT_ELEMENT_NAME);
+        nameIDPolicy.setFormat(null);
+        eidasAuthnRequest.setNameIDPolicy(nameIDPolicy);
```

Comment:
Related to previous null check comment: one more test for condition `eidasAuthnRequest.setNameIDPolicy(null);`